### PR TITLE
School 195&196 tests for authorization and authentication

### PR DESCRIPTION
**Pull request by ArmineZak**

### CODE REVIEW

File: src/test/java/com/epam/edumanagementsystem/security/config/SecurityConfigTest.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+import org.springframework.transaction.annotation.Transactional;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@Transactional
+public class SecurityConfigTest {
+
```
Comment (schoolmanagementproject2022) [inline src/test/java/com/epam/edumanagementsystem/security/config/SecurityConfigTest.java:17]: In this test class you've probably covered all possible redirection scenarios either positive or negative cases. If all tests are passing then this class is good. The only thing is why did you add @Transactional annotation on the class?

File: src/test/java/com/epam/edumanagementsystem/security/rest/api/LoginControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    @Test
+    public void loginUnauthenticated() throws Exception {
+        mockMvc.perform(get("/login"))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    @WithMockUser
+    public void loginAuthenticated() throws Exception {
+        mockMvc.perform(get("/login"))
```
Comment (schoolmanagementproject2022) [inline src/test/java/com/epam/edumanagementsystem/security/rest/api/LoginControllerTest.java:28]: Could you please explain the second scenario?

File: src/test/java/com/epam/edumanagementsystem/security/rest/api/LoginControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+public class LoginControllerTest {
```
Comment (schoolmanagementproject2022) [inline src/test/java/com/epam/edumanagementsystem/security/rest/api/LoginControllerTest.java:15]: This class is good for me

File: src/test/java/com/epam/edumanagementsystem/security/config/AuthenticationTest.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+@AutoConfigureMockMvc
+public class AuthenticationTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Test
+    public void testLoggingWithInvalidUser() throws Exception {
+        mockMvc.perform(formLogin()
+                .user("wrong-user@gmail.com").password( "123456"))
```
Comment (schoolmanagementproject2022) [inline src/test/java/com/epam/edumanagementsystem/security/config/AuthenticationTest.java]: That would be great not to hardcode username and password values, and in general any value, it would be better to have class level Strings. This is regarding to every test class in general

File: src/test/java/com/epam/edumanagementsystem/EdumanagementsystemApplicationTests.java [hunk 5]
```diff
@@ -6,8 +6,5 @@
 @SpringBootTest
 class EdumanagementsystemApplicationTests {
```
Comment (schoolmanagementproject2022) [inline src/test/java/com/epam/edumanagementsystem/EdumanagementsystemApplicationTests.java:7]: Actually didn't get the meaning of this test class

Comment (ArmineZak) [reply src/test/java/com/epam/edumanagementsystem/EdumanagementsystemApplicationTests.java:7]: Neither did I

File: src/test/java/com/epam/edumanagementsystem/security/config/SecurityConfigTest.java (hunk 1 above)
Comment (ArmineZak) [reply src/test/java/com/epam/edumanagementsystem/security/config/SecurityConfigTest.java:17]: There was an issue with /classses/ path due to lazy initialization, which was causing the test to fail, so it was advised to use @Transactional on the class to avoid such behavior.

**Review by schoolmanagementproject2022 — APPROVED** [review]
